mithril-core/src/merkle_tree.rs

```diff
@@ -339,52 +628,84 @@ mod tests {
-                let bytes = pf.to_bytes();
-                let deserialised = Path::from_bytes(&bytes).unwrap();
-                assert!(t.to_commitment().check(&values[i], &deserialised).is_ok());
-
-                let encoded = bincode::serialize(&pf).unwrap();
-                let decoded: Path<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
-                assert!(t.to_commitment().check(&values[i], &decoded).is_ok());
-            })
-        }
+    #[test]
+    fn test_bytes_path((t, values) in arb_tree(30)) {
+        values.iter().enumerate().for_each(|(i, _v)| {
+            let pf = t.get_path(i);
+            let bytes = pf.to_bytes();
+            let deserialised = Path::from_bytes(&bytes).unwrap();
+            assert!(t.to_commitment().check(&values[i], &deserialised).is_ok());
+
+            let encoded = bincode::serialize(&pf).unwrap();
+            let decoded: Path<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
+            assert!(t.to_commitment().check(&values[i], &decoded).is_ok());
+        })
+    }
 
-        #[test]
-        fn test_bytes_tree((t, values) in arb_tree(5)) {
-            let bytes = t.to_bytes();
-            let deserialised = MerkleTree::<Blake2b<U32>>::from_bytes(&bytes).unwrap();
-            let tree = MerkleTree::<Blake2b<U32>>::create(&values);
-            assert_eq!(tree.nodes, deserialised.nodes);
-
-            let encoded = bincode::serialize(&t).unwrap();
-            let decoded: MerkleTree::<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
-            assert_eq!(tree.nodes, decoded.nodes);
-        }
+    #[test]
+    fn test_bytes_tree((t, values) in arb_tree(5)) {
+        let bytes = t.to_bytes();
+        let deserialised = MerkleTree::<Blake2b<U32>>::from_bytes(&bytes).unwrap();
+        let tree = MerkleTree::<Blake2b<U32>>::create(&values);
+        assert_eq!(tree.nodes, deserialised.nodes);
 
-        #[test]
-        fn test_bytes_tree_commitment((t, values) in arb_tree(5)) {
-            let encoded = bincode::serialize(&t.to_commitment()).unwrap();
-            let decoded: MerkleTreeCommitment::<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
-            let tree_commitment = MerkleTree::<Blake2b<U32>>::create(&values).to_commitment();
-            assert_eq!(tree_commitment.root, decoded.root);
+        let encoded = bincode::serialize(&t).unwrap();
+        let decoded: MerkleTree::<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
+        assert_eq!(tree.nodes, decoded.nodes);
+    }
+
+    #[test]
+    fn test_bytes_tree_commitment((t, values) in arb_tree(5)) {
+        let encoded = bincode::serialize(&t.to_commitment()).unwrap();
+        let decoded: MerkleTreeCommitment::<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
+        let tree_commitment = MerkleTree::<Blake2b<U32>>::create(&values).to_commitment();
+        assert_eq!(tree_commitment.root, decoded.root);
+    }
+
```

Comment:
We should also test Ser/Deser of the tree batch commitment. They're different right? And going over the tests, I saw that we don't have tests for failures. We should add one of those for the path verification.